### ISSUE TITLE
Major version release

### ISSUE THREAD

**Issue by jeremymanning** [ContextLab/hypertools#183]
Let's work towards a feature list that we'd want to incorporate into a 1.0.0 release.  For example:
- What's currently broken and/or set up in a sub-optimal way?
- What features would be nice to add?
- What can we do to finalize the API and maximize compatibility with future releases?
- Is there anything we should *remove* from the current list?  (E.g. features we don't support well and that we don't want to support, legacy argument names that we're keeping around for backwards compatibility, etc.)
- Come up with some example use case "stories" and work through (story board) how we're expecting the user to leverage hypertools.  These should showcase each of the major features of hypertools and could be used as running examples (and we could set up tests around them to make sure future changes don't break any of the target use cases).

**Comment by andrewheusser** [comment]
one thing that would be potentially tricky to implement but well worth it would be to rewrite the `transform` method of a geo to transform the input data with the pipeline of models fit when the geo was created, instead of just replicating the pipeline, but fitting new models.

**Comment by jeremymanning** [comment]
agreed...we would need to save the model objects and any inputs they require.  we need a systematic way (pickling?  saving lists or dictionaries of arguments?) of keeping track of all the needed stuff, ideally in a way that generalizes across the various models we might want to use.

**Comment by andrewheusser** [comment]
yes, and also rewriting some of the functions (align coming to mind, but maybe others) to conform to the `fit` `transform` sklearn API

**Comment by jeremymanning** [comment]
'twould be great.  i think the pymvpa2 implementation of hyperalignment would be adaptable to that style [[link](https://github.com/PyMVPA/PyMVPA/blob/master/mvpa2/algorithms/hyperalignment.py)]

**Comment by andrewheusser** [comment]
yes, i think that will work! `train`->`fit`, `transform`->`transform`

**Comment by jeremymanning** [comment]
This planning issue's goals -- and the specific ask in the thread to rewrite `align` (and related tools) around the scikit-learn `fit`/`transform`/`fit_transform` API -- have been addressed in the HyperTools 1.0 update (PR #272, https://github.com/ContextLab/hypertools/pull/272).

All aligners (`Procrustes`, `HyperAlign`, `SharedResponseModel`, `NullAlign`) now subclass a common `Aligner(sklearn.base.BaseEstimator)` with real `fit`/`transform`/`fit_transform` methods:
```python
import numpy as np, hypertools as hyp
a = np.random.randn(10, 3)
b = a @ np.array([[0, -1, 0], [1, 0, 0], [0, 0, 1]]) + 0.01 * np.random.randn(10, 3)
out = hyp.align([a, b], align="Procrustes")
# -> list of 2 arrays, each shape (10, 3)
```
More broadly, this issue's call for a finalized 1.0 API, removed cruft, and consolidated feature set is exactly what the 1.0 refactor delivers. Closing as addressed in the 1.0 update -- please reopen if you hit anything else.
